### PR TITLE
[Draft] Moving contents to manual pages from reference guide

### PR DESCRIPTION
**Pull request by xh4n3**
I'm creating this PR to demonstrate the concept of stage 1 mentioned in #2753 .

I've deleted some sections from the reference guide, and moved them into the manual page. There is a live manual page preview at [my working branch](https://github.com/xh4n3/bpftrace/blob/update-man-page/man/adoc/bpftrace.adoc).

Please leave your valuable feedback on this change, thanks! If there is no objections, I'll go moving the remaining sections shortly.

**Note: The deletions are meant to show which parts are being moved. If we all agree on this change, I'll revert the deletions before merging this.**

### CODE REVIEW

**Comment by danobi** [comment]
If I understand correctly, the plan is to complete the manual, then start deleting from the reference guide (or replace the reference guide)?

File: docs/reference_guide.md [hunk 1]
```diff
@@ -9,137 +9,6 @@ Also, please keep these docs as terse as possible to maintain their brevity (ins
...
-    - [9. `lhist()`: Linear Histogram](#9-lhist-linear-histogram)
-    - [10. `print()`: Print Map](#10-print-print-map)
-- [Output](#output)
-    - [1. `printf()`: Per-Event Output](#1-printf-per-event-output)
-    - [2. `interval`: Interval Output](#2-interval-interval-output)
-    - [3. `hist()`, `printf()`: Histogram Printing](#3-hist-print-histogram-printing)
-    - [4. `SIGUSR1`: On-Demand Output](#4-sigusr1-on-demand-output)
-- [BTF Support](#btf-support)
-- [Advanced Tools](#advanced-tools)
-- [Errors](#errors)
```
Comment (danobi) [inline docs/reference_guide.md:125]: I'm a bit confused about the deleted entries here. It doesn't look like it was moved into the man page. Is that intentional?

Comment (xh4n3) [reply docs/reference_guide.md:125]: Yes, this is intentional.

I'm moving contents those should be included in the man page, and deleting those should not. So when the reference guide become empty, the man page is complete.

**Comment by xh4n3** [comment]
> If I understand correctly, the plan is to complete the manual, then start deleting from the reference guide (or replace the reference guide)?

@danobi Yes, the deletions from this PR is just for easy review, will be reverted before merge.

After merging this, the next stage is to create a new quick start guide and delete the whole reference page.

File: docs/reference_guide.md (hunk 1 above)
Comment (danobi) [reply docs/reference_guide.md:125]: Ah ok, I see. Your intention is to "delete" the table of contents in this PR. But the actual contents will be moved in subsequent PRs.

**Comment by danobi** [comment]
I gave the man pages changes a skim -- they lgtm

**Comment by jordalgo** [comment]
I do find the markdown a bit easier to navigate on github but maybe it's worth deleting it so there is a single source of truth for using bpftrace.

File: man/adoc/bpftrace.adoc [hunk 2]
```diff
@@ -53,104 +105,225 @@ Custom prefix to easily link section
 
 === Output format
 
-*-B* _MODE_::
+==== *-B* _MODE_
   Set the buffer mode for stdout. Valid values are::
```
Comment (viktormalik) [inline man/adoc/bpftrace.adoc:109]: Remove indentation as in the rest of the section?

File: man/adoc/bpftrace.adoc [hunk 3]
```diff
@@ -53,104 +105,225 @@ Custom prefix to easily link section
...
 
     *none* No buffering. Each I/O is written as soon as possible +
     *line* Data is written on the first newline or when the buffer is full.
     This is the default mode. +
     *full* Data is written once the buffer is full.
 
-*-f* _FORMAT_::
+==== *-f* _FORMAT_
   Set the output format. Valid values are::
```
Comment (viktormalik) [inline man/adoc/bpftrace.adoc:117]: Same as above.

**Review by viktormalik — COMMENTED** [review]
Looks good to me, just two nits.

I think that you can go on with converting all the parts missing from the manual.

**Pull request closed without merging**